bridge-impl/src/main/java/com/liferay/faces/bridge/event/internal/HeaderRequestPhaseListener.java

```diff
@@ -38,31 +38,30 @@
 	// serialVersionUID
 	private static final long serialVersionUID = 8470095938465172618L;
 
-	// Protected (Lazy-Initialized) Constants
-	private static Boolean VIEW_PARAMETERS_ENABLED;
-
-	// Private Data Members
-	private PhaseId phaseId = PhaseId.RESTORE_VIEW;
+	// Instance field must be declared volatile in order for the double-check idiom to work (requires JRE 1.5+)
+	private static volatile Boolean viewParametersEnabled;
 
 	@Override
 	public void afterPhase(PhaseEvent phaseEvent) {
 
 		FacesContext facesContext = phaseEvent.getFacesContext();
 
-		if (VIEW_PARAMETERS_ENABLED == null) {
+		// First check without locking (not yet thread-safe)
+		if (viewParametersEnabled == null) {
 
 			synchronized (this) {
```

Comment:
@stiemannkj1 Similar to a `Converter`, I will assume that a `PhaseListener` is effectively stored in application scope since the `PhaseEvent` object has a `getFacesContext()` method. If this is indeed the case, then `synchronized(this)` is sufficient. Just to make sure, please put a `System.err.println` in the PhaseListener constructor and see when (and possibly how many times) it is called.